newIDE/app/public/external/utils/parent-editor-interface.js

```diff
@@ -0,0 +1,51 @@
+let electron = null;
+let remote = null;
+let electronWindow = null;
+let ipcRenderer = null;
+
+if (typeof require === 'function') {
+  electron = require('electron');
+  remote = require('@electron/remote');
+  electronWindow = remote.getCurrentWindow();
+  ipcRenderer = electron.ipcRenderer;
+}
+
+export const setTitle = title => {
+  if (electronWindow) electronWindow.setTitle(title);
+  else document.title = title;
+};
+
+export const closeWindow = () => {
+  if (remote) remote.getCurrentWindow().close();
+  else {
+    sendMessageToParentEditor('close');
+  }
+};
+
+export const sendMessageToParentEditor = (id, payload) => {
+  if (ipcRenderer) {
+    if (payload) ipcRenderer.send(id, payload);
+    else ipcRenderer.send(id);
+  } else if (window && window.opener) {
+    window.opener.postMessage({
+      id,
+      payload,
+    }, '*');
+  } else {
+    console.error(`Unable to send message ${id} to the main editor.`);
+  }
+};
+
+export const onMessageFromParentEditor = (id, callback) => {
+  if (ipcRenderer) {
+    ipcRenderer.on(id, (event, payload) => {
+      callback(payload);
+    });
+  } else {
+    window.addEventListener('message', event => {
```

Comment:
is there a risk this event listener stays listening?
Should there be some listener collector to clean them up sometimes?

Reply:
Very good question, in theory all addEventListener must be removeListener'ed later.
In this case, when the editor is closed, the whole window (in a web-app) or the "BrowserWindow" (in Electron, it's basically the same concept except you have access to more native stuff) is deleted and so all the callbacks/listeners and even the JS engine is deleted.

So it should be all good!